joern-cli/frontends/solidity2cpg/src/main/scala/io/joern/solidity2cpg/passes/AstCreator.scala

```diff
@@ -910,29 +859,30 @@ class AstCreator(filename: String, sourceUnit: SourceUnit, global: Global) exten
     }
     else
       operation.operator match {
+        // TODO Can `NOT` really be postfix?
         case "!"  => Operators.logicalNot
         case "++" => Operators.postIncrement
         case "--" => Operators.postDecrement
         case _    => throw new Exception("Unsupported unary operator: " + operation.operator)
       }
-    val code =
-      if (operation.isPrefix)
-        operation.operator + subExpression.root.map(_.properties(PropertyNames.CODE)).mkString("")
-      else
-        subExpression.root.map(_.properties(PropertyNames.CODE)).mkString("") + operation.operator
-    val callNode = NewCall()
-      .name(operatorName)
-      .methodFullName(operatorName)
-      .dispatchType(DispatchTypes.STATIC_DISPATCH)
-      .code(code)
+
+    // TODO
+    //  - Replace with rootCode
+    //  - Find better default
+    val subExprCode  = subExpression.root.map(_.properties(PropertyNames.CODE)).getOrElse("")
+    val opCode       = operation.operator
+    val code         = if (operation.isPrefix) s"$opCode$subExprCode" else s"$subExprCode$opCode"
+    val lineNumber   = operation.lineNumber.map(Integer.valueOf)
+    val columnNumber = operation.columnNumber.map(Integer.valueOf)
+
+    val callNode = operatorCallNode(operatorName, code = code, line = lineNumber, column = columnNumber)
       .order(order)
       .argumentIndex(order)
-      .columnNumber(operation.columnNumber.get)
-      .lineNumber(operation.lineNumber.get)
 
     Ast(callNode).withChild(subExpression)
   }
   private def astForTypeNameExpression(expression: TypeNameExpression, order: Int): Ast = {
+    // TODO If this really should be empty, remove the method.
```

Comment:
I don't think it should be empty, I didn't get to creating it because in the contracts views it wasn't used yet :)